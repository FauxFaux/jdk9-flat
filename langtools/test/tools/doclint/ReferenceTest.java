/*
 * @test /nodynamiccopyright/
 * @build DocLintTester
 * @run main DocLintTester -Xmsgs:-reference ReferenceTest.java
 * @run main DocLintTester -ref ReferenceTest.out ReferenceTest.java
 */

/** */
public class ReferenceTest {
    /**
     * @param x description
     */
    public int invalid_param;

    /**
     * @param x description
     */
    public class InvalidParam { }

    /**
     * @param x description
     */
    public void param_name_not_found(int a) { }

    /**
     * @param <X> description
     */
    public class typaram_name_not_found { }

    /**
     * @see Object#tooStrong()
     */
    public void ref_not_found() { }

    /**
     * @return x description
     */
    public int invalid_return;

    /**
     * @return x description
     */
    public void invalid_return();

    /**
     * @throws Exception description
     */
    public void exception_not_thrown() { }
}

