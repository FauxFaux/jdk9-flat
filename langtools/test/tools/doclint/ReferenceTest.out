ReferenceTest.java:11: error: invalid use of @param
     * @param x description
       ^
ReferenceTest.java:16: error: invalid use of @param
     * @param x description
       ^
ReferenceTest.java:21: error: @param name not found
     * @param x description
              ^
ReferenceTest.java:23: warning: no @param for a
    public void param_name_not_found(int a) { }
                ^
ReferenceTest.java:26: error: @param name not found
     * @param <X> description
               ^
ReferenceTest.java:31: error: reference not found
     * @see Object#tooStrong()
            ^
ReferenceTest.java:36: error: invalid use of @return
     * @return x description
       ^
ReferenceTest.java:41: error: invalid use of @return
     * @return x description
       ^
ReferenceTest.java:46: error: exception not thrown: java.lang.Exception
     * @throws Exception description
               ^
8 errors
1 warning

